05/ContactForm.js

```diff
@@ -43,111 +10,122 @@ const initialState = {
     subject:'',
     message:'',
 }
-const reducer = (state,{name,value}) => {
-    const newState = {...state}
-    newState[name] = value
-    return newState
+
+const reducer = (state,action) => {
+    const {type, value} = action;
+    if(type === 'reset'){
```

Comment:
ok, ale jeśli pole w formularzu nazywałoby się `reset` to można zdziwić się jego działania :P